src/openfermion/functionals/get_one_norm.py

```diff
@@ -0,0 +1,135 @@
+#   limitations under the License.
+"""
+Function to calculate the 1-Norm of a molecular Hamiltonian
+(in the restricted formalism) after fermion-to-qubit transformation
+from a MolecularData class. See https://arxiv.org/abs/2103.14753
+for more information on the 1-norm.
+"""
+
+import numpy as np
+
+
+def get_one_norm_mol(molecule, no_constant=True):
+    r"""
+
+    Returns the 1-Norm of a RHF or ROHF Hamiltonian described in
+    https://arxiv.org/abs/2103.14753 after a fermion-to-qubit
+    transformation given nuclear constant, one-body (2D np.array)
+    and two-body (4D np.array) integrals in spatial orbital basis.
+
+    Parameters
+    ----------
+
+    molecule : MolecularData class representing a molecular Hamiltonian
+    constant(float) : Nuclear repulsion or adjustment to constant shift in
+        Hamiltonian from integrating out core orbitals
+    one_body_integrals(ndarray) : An array of the one-electron integrals having
+        shape of (n_orb, n_orb), where n_orb is the number of spatial orbitals.
+    two_body_integrals(ndarray) : An array of the two-electron integrals having
+        shape of (n_orb, n_orb, n_orb, n_orb).
+    no_constant (default: True) : If True, do not return the constant term in
+                in the (majorana/qubit) Hamiltonian
+
+    Returns
+    -------
+    one_norm : 1-Norm of the qubit Hamiltonian
+    """
+    if no_constant:
+        return _get_one_norm_woconst(molecule.one_body_integrals,
+                                     molecule.two_body_integrals)
+    else:
+        return _get_one_norm(molecule.nuclear_repulsion,
+                             molecule.one_body_integrals,
+                             molecule.two_body_integrals)
+
+
+def get_one_norm_int(constant,
+                     one_body_integrals,
+                     two_body_integrals,
+                     no_constant=True):
+    """
+
+    Parameters
+    ----------
+    constant(float) : Nuclear repulsion or adjustment to constant shift in
+        Hamiltonian from integrating out core orbitals
+    one_body_integrals(ndarray) : An array of the one-electron integrals having
+        shape of (n_orb, n_orb), where n_orb is the number of spatial orbitals.
+    two_body_integrals(ndarray) : An array of the two-electron integrals having
+        shape of (n_orb, n_orb, n_orb, n_orb).
+    no_constant (default: True) : If True, do not return the constant term in
+                in the (majorana/qubit) Hamiltonian
+
+    Returns
+    -------
+    one_norm : 1-Norm of the qubit Hamiltonian
+
+    """
+    if no_constant:
+        return _get_one_norm_woconst(one_body_integrals, two_body_integrals)
+    else:
+        return _get_one_norm(constant, one_body_integrals, two_body_integrals)
+
+
+def _get_one_norm(constant, one_body_integrals, two_body_integrals):
+    """Compute 1-norm given molecular integrals"""
+    n_orb = one_body_integrals.shape[0]
+
+    htilde = constant
```

Comment:
This should be htilde = abs(constant), right? Otherwise we can wind up with negative 1 norms...

Reply:
No it is actually correct like this. The reason is that you have in principle two (or three if you consider an active space) contributions to the constant term in majorana operators: The nuclear repulsion (always positive), the mean field energy of the frozen orbitals if you consider a CAS (always negative) and the ((1 / 2 * g_[ppqq]) - (1 / 4 *g_[pqpq])) term from rearranging the majorana operators. It is important to actually take the absolute value after counting them up together as they can cancel eachother. I take the absolute value of htilde on the line below, so the 1-norm can never be negative.

Reply:
Ah ok, thanks for the clarification!

Reply:
no I think this goes back to the comment I made in the "issue" about this. It depends on how you are defining and accessing the LCU. There is a way to represent the LCU such that what @Emieeel is saying is true. And it is a clever trick that he's pointed out there. But under the simplest way to simulate the LCU, you don't get that subtraction from using the Majorana operators.